src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp

```diff
@@ -447,16 +466,33 @@ inline void keywordAquifer( SummaryConfig::keyword_list& list,
     .parameterType( parseKeywordType(keyword.name()) )
     .isUserDefined( is_udq(keyword.name()) );
 
-    if (keyword.size() && keyword.getDataRecord().getDataItem().hasValue(0)) {
-        for( const int id: keyword.getIntData()) {
-            if (aquiferConfig.hasAquifer(id)) {
-                list.push_back(param.number(id));
-            } else {
-                handleMissingAquifer(parseContext, errors, keyword.location(), id);
+    const auto is_numeric = is_numeric_aquifer(keyword.name());
+    const auto& pertinentIDs = is_numeric
+        ? numericAquiferIDs : analyticAquiferIDs;
+
+    if ((keyword.size() == decltype(keyword.size()){0}) ||
```

Comment:
No - `decltype` here is over engineering; `if (keyword.size() == 0)` - or alternatively we can add an `empty()` method to the `DeckKeyword` class.

Reply:
> we can add an `empty()` method to the `DeckKeyword` class.

I've made that change in #2474 and rebased this PR on top of that.